ckanext/vocabulary/blueprints/vocabulary.py

```diff
@@ -205,20 +232,37 @@ def new_tags(id):
             tag_dict = {
                 'name': en_tags.strip(),
                 'name_translated-en': en_tags.strip(),
-                'name_translated-ar': ar_tags,
+                'name_translated-ar': ar_tags.strip(),
                 'vocabulary_id': id
             }
+            non_free_tag_key = list(non_free_tags.keys())
+            en_tag = en_tags.lower().strip()
+            ar_tag = ar_tags.lower().strip()
             try:
-                if en_tags.strip() in free_tags or en_tags.strip() in non_free_tags:
+                if en_tag.strip() in free_tags:
                     raise logic.ValidationError(
-                        error_dict= {'name_translated-en': [f'Tag {en_tags.strip()}  already exist']}
+                        error_dict= {'name_translated-en': [f'Tag: "{en_tags.strip()}"  already exist in Free Tags']}
                     )
+                elif en_tag in non_free_tag_key:
+                    tagname = non_free_tags[en_tag]
+                    raise logic.ValidationError(
+                        error_dict= {'name_translated-en': [f'Tag: "{en_tags.strip()}"  already exist in Category: "{tagname}"']}
+                    )
+                elif ar_tag in non_free_tag_key:
+                    tagname = non_free_tags[ar_tag]
+                    vocab_name = tagname['vocab']
+                    trans_name = tagname['tag']
+                    raise logic.ValidationError(
+                        error_dict= {
+                            'name_translated-en': [f'AR Tag: "{ar_tags.strip()}" already exist in Category: {vocab_name} under EN Tag: {trans_name}']
+                            }
+                    )   
                 else:
                     get_action('tag_create')(context, tag_dict)
                     h.flash_success(u"Tag sucessfully added to the category.")
                     return h.redirect_to(h.url_for('vocabulary_read', id=id))
```

Comment:
Also not high priority but if the code same in if and else block cant we just add it in a function and call that function instead of maintaining same piece of code in two places?

Reply:
@MuhammadIsmailShahzad I thought of using function. but doing that might more arguments and the error report is quite different

Reply:
for example in one i was tracking the number of errors and the other i was not

Reply:
ok. Gonna approve and merge this now and we can check the duplication issue later if that is required.